include/open62541/plugin/log.h

```diff
@@ -63,6 +63,8 @@ typedef struct {
     void *context; /* Logger state */
 
     void (*clear)(void *context); /* Clean up the logger plugin */
+
+    bool externalLogger; /* The logger is not deleted with the config */
```

Comment:
Don't add the flag to the plugin itself.
Put it next to the plugin in the config.